library/ssl_tls13_client.c

```diff
@@ -1754,7 +1754,13 @@ static int ssl_client_hello_write_partial( mbedtls_ssl_context* ssl,
 #endif /* MBEDTLS_SSL_ALPN */
 
 #if defined(MBEDTLS_SSL_MAX_FRAGMENT_LENGTH)
-    ssl_write_max_fragment_length_ext( ssl, buf, end, &cur_ext_len );
+    if( ( ret = ssl_write_max_fragment_length_ext( ssl, buf,
+                                                   end - buf, &cur_ext_len )  ) != 0 )
```

Comment:
```suggestion
    if( ( ret = ssl_write_max_fragment_length_ext( ssl, buf,
                                                   (size_t)( end - buf ), 
                                                   &cur_ext_len )  ) != 0 )
```
Some compilers could complain about the implicit signed to unsigned cast otherwise.